bundles/org.openhab.core.model.rule.runtime/src/org/openhab/core/model/rule/runtime/internal/DSLRuleProvider.java

```diff
@@ -317,15 +296,16 @@ private String removeIndentation(String script) {
             s += "\n\n";
         }
         String firstLine = s.lines().findFirst().orElse("");
-        String indentation = firstLine.substring(0, firstLine.length() - firstLine.stripLeading().length());
+        String indentation = firstLine == null ? ""
+                : firstLine.substring(0, firstLine.length() - firstLine.stripLeading().length());
         return s.lines().map(line -> (line.startsWith(indentation) ? line.substring(indentation.length()) : line))
                 .collect(Collectors.joining("\n"));
     }
 
     private @Nullable Trigger mapTrigger(EventTrigger t) {
         if (t instanceof SystemOnStartupTrigger) {
             Configuration cfg = new Configuration();
-            cfg.put("startlevel", 20);
+            cfg.put("startlevel", 40);
```

Comment:
This change fixes the issue for me. I have no idea, why I have ever set it to 20 in the first place.
According to the documentation at https://www.openhab.org/docs/configuration/rules-dsl.html#system-based-triggers, "startlevels less than 40 are not available as triggers". Setting it to 40 will only execute them once during startup.